cmd/swapd/main.go

```diff
@@ -180,6 +181,10 @@ var (
 				Name:  flagDeploy,
 				Usage: "Deploy an instance of the swap contract",
 			},
+			&cli.StringFlag{
+				Name:  flagForwarderAddress,
+				Usage: "If --deploy is set, provide the address of the trusted forwarder contract for the deployed swap contract",
+			},
```

Comment:
Just throwing out an idea. It's probably not even a good idea. If you wanted to simplify the inter-flag dependencies, you could call this flag `--deploy-with-forwarder ADDRESS`.  Although, I guess that would still leave the question if it is an error if `--deploy` and `--deploy-with-forwarder` are passed on the same command line.

If you leave it as-is, maybe change the description to something like:
> Specifies the Ethereum address of the trusted forwarder contract when deploying the swap contract. Ignored if --deploy is not passed.

Reply:
description updated! left the flags as-is for now